tensornetwork/matrixproductstates/mps.py

```diff
@@ -341,6 +431,315 @@ def measure_two_body_correlator(self, op1: Union[BaseNode, Tensor],
+
+    Args:
+      gate: a one-body gate
+      site: the site where the gate should be applied
+      
+    """
+    if len(gate.shape) != 2:
+      raise ValueError('rank of gate is {} but has to be 2'.format(
+          len(gate.shape)))
+    if site < 0 or site >= len(self):
+      raise ValueError('site = {} is not between 0 <= site < N={}'.format(
+          site, len(self)))
+    gate_node = Node(gate, backend=self.backend.name)
+    gate_node[1] ^ self.nodes[site][1]
+    edge_order = [self.nodes[site][0], gate_node[0], self.nodes[site][2]]
+    self.nodes[site] = contract_between(
+        gate_node, self.nodes[site],
+        name=self.nodes[site].name).reorder_edges(edge_order)
+
+  def check_orthonormality(self, which: Text, site: int) -> Tensor:
+    """
+    Check orthonormality of tensor at site `site`.
+
+    Args:
+      which: * if `'l'` or `'left'`: check left orthogonality
+             * if `'r`' or `'right'`: check right orthogonality
+      site:  The site of the tensor.
+    Returns:
+      scalar `Tensor`: The L2 norm of the deviation from identity.
+    Raises:
+      ValueError: If which is different from 'l','left', 'r' or 'right'.
+    """
+    if which not in ('l', 'left', 'r', 'right'):
+      raise ValueError(
+          "Wrong value `which`={}. "
+          "`which` as to be 'l','left', 'r' or 'right.".format(which))
+    n1 = self.nodes[site]
+    n2 = conj(n1)
+    if which in ('l', 'left'):
+      n1[0] ^ n2[0]
+      n1[1] ^ n2[1]
+    elif which in ('r', 'right'):
+      n1[2] ^ n2[2]
+      n1[1] ^ n2[1]
+    result = n1 @ n2
+    return self.backend.norm(
+        abs(result.tensor - self.backend.eye(
+            N=result.shape[0], M=result.shape[1], dtype=self.dtype)))
+
+  def get_node(self, site: int) -> BaseNode:
+    """
+    Returns the `Node` object at `site`.
+    If `site==len(self) - 1` `BaseMPS.connector_matrix`
+    is absorbed fromt the right-hand side into the returned 
+    `Node` object.
+
+    Args:
+      site: The site for which to return the `Node`.
+    Returns:
+      `Node`: The node at `site`.
+    """
+    if site >= len(self):
+      raise IndexError(
+          'index `site` = {} is out of range for len(mps)= {}'.format(
+              site, len(self)))
+    if site < 0:
+      raise ValueError(
+          'index `site` has to be larger than 0 (found `site`={}).'.format(
+              site))
+    if (site == len(self) - 1) and (self.connector_matrix is not None):
+      self.nodes[site][2] ^ self.connector_matrix[0]
+      order = [
+          self.nodes[site][0], self.nodes[site][1], self.connector_matrix[1]
+      ]
+      return contract_between(
+          self.nodes[site],
+          self.connector_matrix,
+          name=self.nodes[site].name,
+          output_edge_order=order)
+    return self.nodes[site]
+
+  def canonicalize(self, normalize: Optional[bool] = True) -> np.number:
+    raise NotImplementedError()
+
+
+class InfiniteMPS(BaseMPS):
```

Comment:
Can you make this (and FiniteMPS) into their own files?

Reply:
can do

Reply:
in fact, it's already supported. I modified the typing to reflect this